lib/spack/spack/test/cmd/ci.py

```diff
@@ -704,27 +714,111 @@ def test_ci_rebuild_basic(tmpdir, mutable_mock_env_path, env_deactivate,
 
     with tmpdir.as_cwd():
         env_cmd('create', 'test', './spack.yaml')
-        with ev.read('test'):
-            root_spec = ('eJyNjsGOwyAMRO/5Ct96alRFFK34ldUqcohJ6BJAQFHUry9Nk66'
-                         'UXNY3v5mxJ3qSojoDBjnqTGelDUVRQZlMIWpnBZya+nJa0Mv1Fg'
-                         'G8waRcmAQkimkHWxcF9NRptHyVEoaBkoD5i7ecLVC6yZd/YTtpc'
-                         'SIBg5Tr/mnA6mt9qTZL9CiLr7trk7StJyd/F81jKGoqoe2gVAaH'
-                         '0uT7ZwPeH9A875HaA9MfidHdHxgxjgJuTGVtIrvfHGtynjkGyzi'
-                         'xRrkHy94t1lftvv1n4AkVK3kQ')
-
-            # Create environment variables as gitlab would do it
-            set_env_var('CI_PROJECT_DIR', working_dir.strpath)
-            set_env_var('SPACK_SIGNING_KEY', signing_key)
-            set_env_var('SPACK_ROOT_SPEC', root_spec)
-            set_env_var('SPACK_JOB_SPEC_PKG_NAME', 'archive-files')
-            set_env_var('SPACK_COMPILER_ACTION', 'NONE')
-            set_env_var('SPACK_CDASH_BUILD_NAME', '(specs) archive-files')
-            set_env_var('SPACK_RELATED_BUILDS_CDASH', '')
-
-            rebuild_output = ci_cmd(
-                'rebuild', fail_on_error=False, output=str)
-
-            print(rebuild_output)
+        with ev.read('test') as env:
+            with env.write_transaction():
+                env.concretize()
+                env.write()
+
+            if not os.path.exists(env_dir.strpath):
+                os.makedirs(env_dir.strpath)
+
+            shutil.copyfile(env.manifest_path,
+                            os.path.join(env_dir.strpath, 'spack.yaml'))
+            shutil.copyfile(env.lock_path,
+                            os.path.join(env_dir.strpath, 'spack.lock'))
+
+            root_spec_build_hash = None
+            job_spec_dag_hash = None
+
+            for h, s in env.specs_by_hash.items():
+                print('{0} -> {1}'.format(s.name, h))
+                if s.name == 'archive-files':
+                    root_spec_build_hash = h
+                    job_spec_dag_hash = s.dag_hash()
+
+            assert root_spec_build_hash
+            assert job_spec_dag_hash
+
+    def fake_cdash_register(build_name, base_url, project, site, track):
+        return ('fakebuildid', 'fakestamp')
+
+    monkeypatch.setattr(ci, 'register_cdash_build', fake_cdash_register)
+
+    print('config fixture: {0}'.format(configuration_dir.strpath))
+    site_config_path = os.path.join(configuration_dir.strpath, 'site')
+
+    print('site configuration_dir files:')
+    for f in os.listdir(site_config_path):
+        print('  {0}'.format(f))
+
+    site_config_yaml_path = os.path.join(site_config_path, 'config.yaml')
+    with open(site_config_yaml_path) as fd:
+        print('config.yaml contents:')
+        print(fd.read())
+
+    # test_configs = os.path.join(spack_paths.test_path, 'data', 'config')
+    monkeypatch.setattr(spack.cmd.ci, 'CI_REBUILD_INSTALL_BASE_ARGS', [
+        '-d', '-k', '-v', '-m', '-C', site_config_path
```

Comment:
Have you confirmed that this appropriately mocks the install_tree? That's  the only thing  I can think  of that I don't see how it's being mocked at this point.

Reply:
This is the last thing I need to look into, which I'm doing now.  But if you don't have any other requested changes, maybe this doesn't have to hold up the PR.

Reply:
This looks good now.